packages/amplify-category-function/src/index.ts

```diff
@@ -111,10 +110,28 @@ export async function initEnv(context) {
   await sequential(functionTasks);
 }
 
-export function invoke(options) {
-  invokeFunction(options);
+// returns a function that can be used to invoke the lambda locally
+export async function getInvoker(context: any, resourceName: string): Promise<(request: InternalInvocation) => Promise<any>> {
+  const breadcrumbs: FunctionBreadcrumbs = context.amplify.readBreadcrumbs(context, categoryName, resourceName);
+  const runtimeManager: FunctionRuntimeLifecycleManager = await context.amplify.loadRuntimePlugin(context, breadcrumbs.pluginId);
+
+  const lambdaConfig = getAllLambdaFunctions(context, context.amplify.pathManager.getBackendDirPath()).find(
+    lambda => lambda.name === resourceName,
+  );
+  return async request =>
+    await runtimeManager.invoke({
+      handler: lambdaConfig.handler,
+      event: request.event,
+      srcRoot: lambdaConfig.basePath,
+      env: {
```

Comment:
we already talked about env vs. envVars thing, that needs to be addressed. Also CLI checked out env name must be available for function invocation as well because of possible rebuilds

Reply:
interface has been updated